recipe/CMakeLists.txt

```diff
@@ -0,0 +1,61 @@
+cmake_minimum_required (VERSION 3.8.0)
+project (libcunit C)
+
+set(HEADERS_DIR  "${PROJECT_SOURCE_DIR}/CUnit/Headers")
+set(SOURCES_DIR "${PROJECT_SOURCE_DIR}/CUnit/Sources")
+set(EXAMPLES_DIR "${PROJECT_SOURCE_DIR}/Examples")
+
+if(MSVC)
+  add_compile_options(/W3 /wd4005 /wd4996 -D_CRT_SECURE_NO_WARNINGS)
+endif()
+
+configure_file(
+  "${HEADERS_DIR}/CUnit.h.in"
+  "${PROJECT_BINARY_DIR}/CUnit.h"
+)
+
+file(READ "${PROJECT_BINARY_DIR}/CUnit.h" CUNIT_H)
+if (BUILD_SHARED_LIBS)
+  string(REPLACE "ifdef CU_DLL" "if 1" CUNIT_H "${CUNIT_H}")
+else()
+  string(REPLACE "ifdef CU_DLL" "if 0" CUNIT_H "${CUNIT_H}")
+endif()
+file(WRITE "${PROJECT_BINARY_DIR}/CUnit.h" "${CUNIT_H}")
+
+include_directories("${PROJECT_BINARY_DIR}")
+include_directories(${HEADERS_DIR})
+
+add_library(libcunit
```

Comment:
As cunit is required by Cyclone DDS and is searched in https://github.com/eclipse-cyclonedds/cyclonedds/blob/48ad8dfe7b10ccf07c83f4394e2b23fc75ce9627/cmake/Modules/FindCUnit.cmake#L41 , I would ensure that the library is called `cunit` on Windows instead of libcunit, as otherwise the `find_library(CUNIT_LIBRARY cunit)` will fail on Windows (as the `lib` prefix is only added in the search process on Unix, not on Windows). 

More in general, if you are interested in this for ROS2/Cyclone DDS compatibility, we could look into using the same CMake script they use for ROS2 cunit builds: https://github.com/ros2/choco-packages/blob/3c6b2741cfec72487257d6a064e886cb6245d2b3/package/cunit/build/CMakeLists.txt .

Reply:
Thanks, this is an extremely helpful comment!

Reply:
Will use the ROS cmake then